Switch indexer fully to mount config approach

ognjenkurtic opened the pull request:
Wrapping up remaining work for allowing the indexer to support additional chains with a simple config change. This PR goes together with[ this one ](https://github.com/omni-network/xchain-indexer/pull/18)from x-indexer repo.

Changes:

- Removing  xchain_config_file_name as now only a single config file will be mounted
 
- Removing omni_config.omni_rpc as it was used for  testing
 
- Renaming syncInterval to sync_interval for consistency
 
- Renaming omni.config.portal_addr to omni_config.omni_predeploy_addr to better reflect the variable usage

From my understanding of TF, this would be a breaking change. If both x-indexer docker images (staging and testnet) are not updated with the results of[ this PR](https://github.com/omni-network/xchain-indexer/pull/18), we would have failure of the deployment for the environment where the image is not applied.


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205273574816497

ognjenkurtic committed:
remove config-file and omni-rpc as flags from docker_compose_xchain tf template

ognjenkurtic committed:
remove config_file_name and omni_config vars from xchain_settings variable

ognjenkurtic committed:
rename syncInterval -> sync_interval; Rename omni_config.portal_addr -> omni_confi.omni_predeploy_addr in main.tf locals

ttarsi reviewed (changes requested):
Good cleanup overall, but same comments as on the other PR, let's keep the config file parameterized

ttarsi commented on aws/templates/docker_compose_xchain.tftpl:
```diff
@@ -15,8 +14,6 @@ services:
             - /bin/indexer
             - index
-            - --config-file=/config/${xchain_config_file_name}.json
-            - --omni-rpc=${omni_rpc}
```
is this flag currently used in staging?

ognjenkurtic replied:
No.

ognjenkurtic replied:
yes, with latest deployed tag 0.1.2

ttarsi replied:
is it overwriting the value in the config file? Or just ignored

ognjenkurtic replied:
Ignored

ognjenkurtic committed:
Add config_file_path to TF config for xchain indexer and make sure it is proprely mounted for  xchain indexer and api instances

ognjenkurtic commented on aws/templates/docker_compose_xchain.tftpl:
```diff
@@ -27,6 +26,8 @@ services:
                   DB_PASSWORD: ${postgres_password}
             volumes:
-            - /config/config.json:/config/config.json
+            - type: bind
```
Had to change the way file is mounted because the path could be non-existent in the container

ognjenkurtic commented:
> Good cleanup overall, but same comments as on the other PR, let's keep the config file parameterized

Thanks. Reverted to a config file path variable and changed the way config file is mounted to allow for container path creation.

ttarsi commented on aws/templates/docker_compose_xchain.tftpl:
```diff
@@ -27,6 +26,8 @@ services:
             volumes:
-            - /config/config.json:/config/config.json
+            - type: bind
+            source: /xchain-indexer-config-to-mount.json
```
maybe it's better to use underscores for file names? hyphens sometimes cause problems

ognjenkurtic replied:
Fixed

ttarsi reviewed (approved):
Left a couple comments

ognjenkurtic committed:
Switch to snake case for file name of the temp file used to mount indexer config in init_script template

ttarsi reviewed (approved):
nice and simple. love it.
